### PR TITLE
Fix tags sort by creation time (descending) on branch/tag dropdowns 

### PR DESCRIPTION
**Pull request by HesterG**
This PR fixes the tags sort issue mentioned in #23432
The tags on dropdown shoud be sorted in descending order of time but are not. Because when getting tags, it execeutes `git tag sort --sort=-taggerdate`. Git supports two types of tags: lightweight and annotated, and `git tag sort --sort=-taggerdate` dosen't work with lightweight tags, which will not give correct result. This PR add `GetTagNamesByRepoID ` to get tags from the database so the tags are sorted. 

Also adapt this change to the droplist when comparing branches. 

Dropdown places:

截屏2023-03-15 14 25 39

截屏2023-03-15 14 25 27

### CODE REVIEW

**Comment by wolfogre** [comment]
I think it should be mentioned in the description that "Git supports two types of tags: lightweight and annotated, and `git tag sort --sort=-taggerdat` dosen't work with lightweight tags."

**Review by wolfogre — APPROVED** [review]
👍

**Review by lunny — APPROVED** [review]

File: modules/context/repo.go [hunk 1]
```diff
@@ -660,20 +660,9 @@ func RepoAssignment(ctx *Context) (cancel context.CancelFunc) {
 		return
 	}
 
-	tags, err := ctx.Repo.GitRepo.GetTags(0, 0)
+	tags, err := repo_model.GetTagNamesByRepoID(ctx, ctx.Repo.Repository.ID)
```
Comment (delvh) [inline modules/context/repo.go:663]: :frowning_face: yet another call to `models` inside `modules`.

File: models/repo/release.go [hunk 2]
```diff
@@ -253,6 +253,28 @@ func GetReleasesByRepoID(ctx context.Context, repoID int64, opts FindReleasesOpt
...
+		ListAll: true,
+	}
+	opts := FindReleasesOptions{
+		ListOptions:   listOptions,
+		IncludeDrafts: true,
+		IncludeTags:   true,
+		HasSha1:       util.OptionalBoolTrue,
+	}
+
+	tags := make([]string, 0)
```
Comment (delvh) [inline models/repo/release.go:268]:
```suggestion
	tags := make([]string)
```

**Review by delvh — APPROVED** [review]

**Comment by jolheiser** [comment]
Agh sorry @delvh I guess my page never updated with your comments, missed it 🙁

**Pull request merged**